x/evm/keeper/state_transition.go

```diff
@@ -242,29 +224,21 @@ func (k *Keeper) ApplyMessage(evm *vm.EVM, msg core.Message, cfg *params.ChainCo
 		return nil, stacktrace.Propagate(sdkerrors.Wrap(types.ErrVMExecution, vmErr.Error()), "vm execution failed")
 	}
 
+	gasUsed := msg.Gas() - leftoverGas
```

Comment:
the `leftoverGas` is updated on `RefundGas` so this value won't be the final one. We should maybe return the used gas too in `RefundGas`

Reply:
that's correct.
I've add a fix which take account of the updated value returned by 'RefundGas'